scripts/globals/utils.lua

```diff
@@ -447,6 +380,73 @@ local relics =
 -- 10 = 119 III (ammo dispensing)
 -------------------------------------------------------
 function utils.getRelicStages(player, relic)
```

Comment:
How do you actually pass a meaningful relic parameter into this function, though?

Reply:
Yeah this doesn't work for me unless I use strings.
```["SPHARAI"] =```
```utils.getRelicStages(player, "SPHARAI")```
etc

Reply:
> How do you actually pass a meaningful relic parameter into this function, though?

The way he wrote it, you don't. the function returns a bitmask of which version of a relic they have. so you don't specify which aegis you are looking for. you get a bitmask of which aegis they they have.

Reply:
> Yeah this doesn't work for me unless I use strings.
> `["SPHARAI"] =`
> `utils.getRelicStages(player, "SPHARAI")`
> etc
> or get rid of the relics table for a bunch of smaller tables
> 
> ```
> function utils.getRelicStages(player, relic)
>         SPHARAI =
>         {
>             18264, 18265, 18637, 18651, 18665, 19746, 19839, 20480, 20481, 21509
>         }
> ...
> for i, id in pairs(relic) do
> ```

the index is already technically a string and I forgot that

Reply:
@cocosolos man I am such a moron lately. it _was_ fine as you had it written and the problem I was looking at only happens after converting it to dsp.whatever.whatever _which the same table name_

Reply:
and then at the last minute I decided it was better off local _anyway_

Reply:
still think this should probably be
```lua
function utils.hasRelic(player, relic, stage)
    SPHARAI =
    {
        18264, 18265, 18637, 18651, 18665, 19746, 19839, 20480, 20481, 21509
    }
...
    if stage ~= nil then
        if player:hasItem(relic[stage]) then
            return true
        else
            return false
        end
    end
    for i, id in pairs(relic) do
        if player:hasItem(id) then
            return true
        end
    end
    return false
end
```